gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```
@@ -1103,6 +1151,124 @@ protected List<Expr> createOperationsStubInitExpr(
+   */
+  private Map<String, HttpRule> parseCustomHttpRules(
+      GapicContext context, Predicate<HttpRule> predicate) {
+    com.google.api.Service service = context.serviceYamlProto();
+    if (service == null || service.getHttp() == null) {
+      return ImmutableMap.of();
+    }
+    return service.getHttp().getRulesList().stream()
+        .filter(predicate)
+        .collect(Collectors.toMap(HttpRule::getSelector, x -> x));
+  }
+
+  /**
+   * Parse the HttpRule for the HttpVerb containing the custom URI. Concat the Http Verb to return
+   * the setting function name.
+   *
+   * <p>This ignores the `Custom` HttpRule. Operation proto only supports GET, POST, DELETE and
+   * Custom requires special logic.
+   *
+   * @param httpRule HttpRule to parse
+   * @return HttpRule.Builder's setter function name when setting for an HttpVerb
+   */
+  private String getHttpVerbSetterFromHttpRule(HttpRule httpRule) {
+    switch (httpRule.getPatternCase().getNumber()) {
+      case 2:
+        return "get";
+      case 3:
+        return "put";
+      case 4:
+        return "post";
+      case 5:
+        return "delete";
+      case 6:
+        return "patch";
+      default:
+        return null;
+    }
+  }
+
+  /**
+   * Parse the HttpRule for the HttpVerb containing the custom URI.
+   *
+   * <p>This ignores the `Custom` HttpRule. Operation proto only supports GET, POST, DELETE and
+   * Custom requires special logic.
+   *
+   * @param httpRule HttpRule to parse
+   * @return Custom HttpRule's URI value
+   */
+  private String getURIValueFromHttpRule(HttpRule httpRule) {
+    switch (httpRule.getPatternCase().getNumber()) {
+      case 2:
+        return httpRule.getGet();
+      case 3:
+        return httpRule.getPut();
+      case 4:
+        return httpRule.getPost();
+      case 5:
+        return httpRule.getDelete();
+      case 6:
+        return httpRule.getPatch();
+      default:
+        return null;
```

This condition should never happen, but in case it happens, can we throw a `IllegalArgumentException` instead of returning null and getting a `NullPointerException` instead?

---

Yeah makes sense. Since this is really only meant for handling Operations, do you think I should also only catch for GET, POST, DELETE and throw `IllegalArgumentException` for any other HttpVerb?

---

Yes, I agree.